src/gen_mod.erl

```diff
@@ -302,6 +303,20 @@ set_module_opt(Host, Module, Opt, Value) ->
                                {#ejabberd_module.opts, Updated})
     end.
 
+%% @doc Non-atomic! You have been warned.
+-spec set_module_opts(ejabberd:server(), module(), _Opts) -> boolean().
```

Comment:
It's a new function.
Can we have a test that checks that the function works correctly?
Description is nice to have too, i.e. how it works, what is Opts and what is the result.

Reply:
Does it replace all module opts or appends opts? - it's what I want to know from description